### PR TITLE
Loop

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#6 

### Changed
- `loop` parameter added to `play` function

### CODE REVIEW

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Thanks for your efforts 💯 
This implementation cannot be stopped in async mode. One of the best solutions is to handle the loop in NavaThread

I suggest that you consider adding this particular feature in a future release that comes after version 0.3.

File: nava/functions.py [hunk 1]
```diff
@@ -82,28 +82,35 @@ def quoter(sound_path, *args, **kwargs):
...
+    :type loop: bool
     :return: None or sound id
     """
     import winsound
-    play_flags = winsound.SND_FILENAME | (async_mode & winsound.SND_ASYNC)
+    play_flags = \
+        winsound.SND_FILENAME | \
+        (async_mode & winsound.SND_ASYNC)
+    if loop:
+        play_flags = play_flags | winsound.SND_LOOP
```
Comment (sadrasabouri) [inline nava/functions.py:102]: @sepandhaghighi What was the problem with following?
```python
    play_flags = \
        winsound.SND_FILENAME | \
        (async_mode & winsound.SND_ASYNC) | \
        (loop & winsound.SND_LOOP)
```

File: nava/params.py [hunk 2]
```diff
@@ -12,6 +12,7 @@
 SOUND_FILE_EXIST_ERROR = "Given sound file doesn't exist."
 SOUND_FILE_PATH_TYPE_ERROR = "Sound file's path should be a string."
 SOUND_ID_EXIST_ERROR = "Given sound id doesn't exist."
+LOOP_ASYNC_ERROR = "The loop option can only be used in async mode."
```
Comment (sadrasabouri) [inline nava/params.py:15]: @sepandhaghighi 
I think we should use a more direct error message, pointing out to the exact variable names. Something like below:
```python
LOOP_ASYNC_ERROR = "`loop` can not be set true when `async_mode` is False."
```

File: nava/thread.py [hunk 3]
```diff
@@ -27,23 +29,34 @@ def run(self):
...
-                self.play_process.terminate()
-                self.play_process.wait()
+                try:
+                    self.play_process.stdout.close()
+                    self.play_process.stdin.close()
+                    self.play_process.stderr.close()
+                    self.play_process.kill()
+                    self.play_process.terminate()
+                except Exception:
+                    _ = 2
```
Comment (sadrasabouri) [inline nava/thread.py:60]: @sepandhaghighi 
What is this :raised_eyebrow:?

**Comment by sadrasabouri** [comment]
@sepandhaghighi I added the `loop` flag to `NavaThread.__init__`. I think that's the better option comparing to setter/getter approach.

File: nava/params.py (hunk 2 above)
Comment (sepandhaghighi) [reply nava/params.py:15]: Good point, I will hande it

Comment (sepandhaghighi) [reply nava/params.py]: fix in https://github.com/openscilab/nava/pull/31/commits/33d1f5c7d5e6d3f739000af85cc75a29c9966e0c

File: nava/functions.py (hunk 1 above)
Comment (sepandhaghighi) [reply nava/functions.py:102]: I'm not sure what the issue is, but it didn't work in this mode.

**Comment by sepandhaghighi** [comment]
@sadrasabouri Could you please test this version on different operating systems? Once confirmed, we can proceed with the final review and merge.

**Comment by sadrasabouri** [comment]
## Local tests on OSs
- [ ] MacOS Sonoma 14.2.1
- [ ] Windows 10
- [ ] Ubuntu 16.04
- [ ] Ubuntu 22.04

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**